src/ReactiveSocket.cpp

```diff
@@ -59,9 +62,14 @@ ReactiveSocket::ReactiveSocket(
       keepaliveTimer_(std::move(keepaliveTimer)),
       resumeSocketListener_(resumeListener) {}
 
+folly::Executor& ReactiveSocket::defaultExecutor() {
```

Comment:
I didnt find any class like this which would give us the checks whether we are on the right thread.
I would implement such checks in later diff. I assume those would be only debug checks.